src/completions.jl

```diff
@@ -299,51 +299,50 @@ end
 
 function global_completions!(items::Dict{String, CompletionItem}, state::ServerState, uri::URI, params::CompletionParams)
     pos = params.position
-    is_macro_invoke = false
-    let context = params.context
-        if !isnothing(context)
-            # Don't trigger completion just by typing a numeric character:
-            context.triggerCharacter in NUMERIC_CHARACTERS && return nothing
-            is_macro_invoke = context.triggerCharacter == "@"
-        end
-    end
-    mod = find_file_module!(state, uri, pos)
     fi = get_fileinfo(state, uri)
-    if !is_macro_invoke && !isnothing(fi)
-        # check if the cursor is within the macro name context `@xxx|`
-        curbyte = xy_to_offset(fi, pos)
-        sn = JS.build_tree(JS.SyntaxNode, fi.parsed_stream)
-        ancestors = byte_ancestors(sn, curbyte-1)
-        if !isempty(ancestors)
-            curnode = first(ancestors)
-            is_macro_invoke = JS.kind(curnode) === K"MacroName"
-        end
+    fi === nothing && return nothing
+    mod = find_file_module!(state, uri, pos)
+    current_token_idx = get_current_token_idx(fi, pos)
+    current_token = fi.parsed_stream.tokens[current_token_idx]
+    current_kind = JS.kind(current_token)
+
+    if current_kind === JS.K"@"
+        edit_start_pos = offset_to_xy(fi, fi.parsed_stream.tokens[current_token_idx - 1].next_byte)
+        is_macro_invoke = true
+    elseif current_kind === JS.K"MacroName"
+        edit_start_pos = offset_to_xy(fi, fi.parsed_stream.tokens[current_token_idx - 2].next_byte)
+        is_macro_invoke = true
+    elseif current_kind === JS.K"TOMBSTONE"
```

Comment:
What's `K"TOMBSTONE"` and why is it necessary?

Reply:
614679a